pkg/cmd/sponsors/list/list_test.go

```diff
@@ -0,0 +1,127 @@
+package list
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCmdList(t *testing.T) {
+	tests := []struct {
+		name  string
+		cli   string
+		wants string
+	}{
+		{
+			name:  "happy path",
+			cli:   "octocat",
+			wants: "octocat",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ios, _, _, _ := iostreams.Test()
+			f := &cmdutil.Factory{
+				IOStreams: ios,
+			}
+
+			argv, err := shlex.Split(tt.cli)
+			assert.NoError(t, err)
+
+			var gotOpts *ListOptions
+			cmd := NewCmdList(f, func(opts *ListOptions) error {
+				gotOpts = opts
+				return nil
+			})
+			cmd.SetArgs(argv)
+			cmd.SetIn(&bytes.Buffer{})
+			cmd.SetOut(&bytes.Buffer{})
+			cmd.SetErr(&bytes.Buffer{})
+
+			_, err = cmd.ExecuteC()
+			assert.NoError(t, err)
+
+			assert.Equal(t, tt.wants, gotOpts.Username)
+		})
+	}
+}
+
+func Test_listRun(t *testing.T) {
+	tests := []struct {
+		name  string
+		opts  *ListOptions
+		wants []string
+	}{
+		{
+			name: "happy path",
+			opts: &ListOptions{
+				HttpClient: func() (*http.Client, error) {
+					r := &httpmock.Registry{}
+
+					r.Register(
+						httpmock.GraphQL(`query SponsorsList\b`),
+						httpmock.StringResponse(`{"data": {"user": {"sponsors": {"totalCount": 2, "nodes": [{"login": "mona"}, {"login": "lisa"}]}}}}`))
+
+					return &http.Client{Transport: r}, nil
+				},
+
+				Username: "octocat",
+				Sponsors: []string{},
+			},
+			wants: []string{"mona", "lisa"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			err := listRun(tt.opts)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.wants, tt.opts.Sponsors)
+		})
+	}
+}
+
+func Test_formatTTYOutput(t *testing.T) {
+	tests := []struct {
+		name  string
+		opts  *ListOptions
+		wants []string
+	}{
+		{
+			name: "Simple table",
+			opts: &ListOptions{
+				Sponsors: []string{"mona", "lisa"},
+			},
+			wants: []string{
+				"SPONSORS",
+				"------",
```

Comment:
I think you mentioned the test was failing before. The underline is stylistic, it's not actually characters on a new line. The styling is handled by ANSI color codes which you can read more about here: https://jafrog.com/2013/11/23/colors-in-terminal.html

Reply:
It was panicking, so I wasn't even getting this check yet. I think your above comment may be the fix I need 👍